.github/workflows/deploy_staging.yml

```diff
@@ -0,0 +1,135 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/SchemaStore/schemastore/master/src/schemas/json/github-workflow.json
+---
+name: Deploy CloudFormation stacks for staging environment
+
+on:
+    pull_request:
+        types:
+            - opened
+            - synchronize
+            - unlabeled
+
+concurrency:
+    group: ${{ github.ref }}
+    cancel-in-progress: true
+
+permissions:
+    id-token: write
+    contents: read
+    pull-requests: write
+    deployments: write
+    repository-projects: write
+
+env:
+    HUSKY: 0
+    NODE_ENV: staging
+    PR_NUM: ${{ github.event.pull_request.number }}
+    HOSTED_ZONE_ID: ${{ secrets.HOSTED_ZONE_ID }}
+    CERTIFICATE_ARN: ${{ secrets.CERTIFICATE_ARN }}
+    MONGODB_URI_PROD: ${{ secrets.MONGODB_URI_PROD }}
+
+    # Turborepo credentials.
+    TURBO_API: ${{ vars.TURBO_API }}
+    TURBO_TOKEN: ${{ secrets.TURBO_TOKEN }}
+    TURBO_TEAM: team_antalmanac
+
+    # AWS credentials.
+    AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+    AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+    AWS_DEFAULT_REGION: us-east-1
+
+jobs:
+    get_staging_configuration:
+        name: Get staging configuration
+
+        runs-on: ubuntu-latest
+
+        outputs:
+            # A staging backend should only be deployed if either the backend or CDK changed.
+            should-deploy-backend: ${{ steps.changed.outputs.backend }} == true ||  ${{ steps.changed.outputs.cdk }} == true
+
+            # If a staging backend is deployed, then the frontend should use its URL. 
+            # Otherwise use the default development API endpoint.
+            antalmanac-api-endpoint: |
+                (${{ steps.changed.outputs.backend }} == true ||  ${{ steps.changed.outputs.cdk }} == true) 
+                && https://staging-${{ github.event.pull_request.number }}.api.antalmanac.com
+                || https://dev.api.antalmanac.com
+
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v3
+
+            - name: Check for changes
+              id: changed
+              uses: dorny/paths-filter@v2
+              with:
+                  filters: |
+                      backend:
+                        - 'apps/backend/**'
+                      cdk:
+                        - 'apps/cdk/**'
+
+    deploy_staging_frontend:
+        name: Deploy staging frontend CloudFormation stack and environment
+
+        needs: get_staging_configuration
+
+        runs-on: ubuntu-latest
+
+        environment:
+            name: staging-frontend-${{ github.event.pull_request.number }}
+            url: https://staging-${{ github.event.pull_request.number }}.antalmanac.com
```

Comment:
What's the reason to use an environment and the comment staging url?

Reply:
It seems like the most correct way to indicate a deployment. Like I'm pretty sure commenting the staging URL is literally just a scuffed version of a real deployment. PPA's implemented deployments properly; you just click on "view deployment"
![image](https://github.com/icssc/AntAlmanac/assets/66761366/08dd9d8d-57fe-41db-bb56-c9bf93a9d6ed)

Reply:
I did the commenting because I couldn't get the environments to delete before but also like I think we literally aren't using any feature of environments besides indicating deploys so I think it just ends up clogging stuff so I don't know if it's worth it 🤷. That being said, I can see it either way so I will just @EricPedley for his opinion.